packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -293,27 +297,31 @@ class Auth {
      */
     async ready() {
         if (this.fetchedToken && this.fetchedToken !== '') {
+            console.log('--- fetched token:', this.fetchedToken)
             const {isGuest, customerId, usid} = this.parseSlasJWT(this.fetchedToken)
             this.set('access_token', this.fetchedToken)
             this.set('customer_id', customerId)
             this.set('usid', usid)
             this.set('customer_type', isGuest ? 'guest' : 'registered')
-            this.pendingToken = Promise.resolve(this.data)
-            return this.pendingToken
+            return Promise.resolve(this.data)
         }
+        // TODO: in my scenario, the code exits early here. It should continue on and refresh the token.
```

Comment:
are we lacking tests or is the tests not written properly to catch this? either way, please add test for this bug, thank you!!! 🙇

Reply:
That's a good point. I'm not going to make a _new_ test because the existing ones are comprehensive already. 

It's just that in the testing of expired token, the bug was not caught because the test did not simulate `pendingToken` being set when initially the token was valid.

I have updated the existing tests to verify:
- `pendingToken` is no longer being set from multiple places.. only `queueRequest()` is supposed to set and delete it.
- the refresh-token logic by simulating real-world scenario (initially having a valid token, and then with expired token)